aether-ui/aether/ui/api/utils.py

```diff
@@ -216,167 +188,130 @@ def is_object_linked(kernel_refs, object_name, entity_type_name=''):
-        else:
-            get_by_name = kernel_data_request(f'projects/?name={projectname}')['results'][0]
-            pipeline.kernel_refs['project'] = get_by_name['id']
-            outcome['successful'].append('Existing {} project used'.format(projectname))
-
-    for entity_type in pipeline.entity_types:
-        schema_name = entity_type['name']
-        schema_data = {
-            'revision': str(uuid.uuid4()),
-            'name': schema_name,
-            'type': entity_type['type'],
-            'definition': entity_type
-        }
+        projects = kernel_data_request(f'projects/?name=Aux')['results']
+        aux_project = projects[0] if len(projects) else {}
+        project_id = aux_project.get('id', str(uuid.uuid4()))
+        project_name = aux_project.get('name', 'Aux')
+    except Exception as e:
+        print(f'Error: {str(e)}')
+    mappingset = {}
+    if pipeline.mappingset:
         try:
-            create_new_kernel_object('schema', pipeline, schema_data, projectname)
-            outcome['successful'].append('{} schema created'.format(
-                entity_type['name']))
+            mappingset = kernel_data_request(f'mappingsets/{pipeline.mappingset}/')
+            project_id = mappingset.get('project')
+            mappingset['name'] = pipeline.name
+            mappingset['input'] = pipeline.schema
         except Exception as e:
-            if overwrite:
-                outcome = overwrite_kernel_schema(pipeline, schema_name, schema_data, projectname, outcome)
-            else:
-                outcome['error'].append(str(e))
-
-    mapping = [
-        [rule['source'], rule['destination']]
-        for rule in pipeline.mapping
+            print(str(e))
+            mappingset = {
+                'id': str(pipeline.mappingset),
+                'name': pipeline.name,
+                'input': pipeline.schema,
+            }
+    else:
+        mappingset = {
+            'id': str(uuid.uuid4()),
+            'name': pipeline.name,
+            'input': pipeline.schema,
+        }
+    mappingsets.append(mappingset)
+    [
+        schemas.append({
+            'name': schema.get('name'),
+            'type': schema.get('type'),
+            'definition': schema,
+            'id': objects_to_overwrite.get('schema', {}).get(schema.get('name'))
+        })
+        for schema in contract.entity_types
     ]
-    mapping_data = {
-        'name': pipeline.name,
+    mappings.append({
+        'id': objects_to_overwrite.get('mapping', {}).get(contract.name, str(contract.id)),
+        'name': contract.name,
         'definition': {
-            'entities': pipeline.kernel_refs.get('projectSchema', {}),
-            'mapping': mapping
-            },
-        'revision': str(uuid.uuid4()),
-        'project': pipeline.kernel_refs['project']
-    }
+            'mapping': [
+                [rule['source'], rule['destination']]
+                for rule in contract.mapping
+            ]
```

Comment:
if you are assigning ids for the mapping sets, why don't you do the same for the schema/project schema and set the `entities` property here? 

Wouldn't that simplify the whole process?

Reply:
And why don't you do it here to avoid further problems in the other side (AKA Kernel).

Reply:
This is the time to redesign the data model to make this as simpler as possible. Really, take your time and if you feel that you need to break something PLEASE GO AHEAD!

Something in the air tells me that the Kernel mapping id and the UI contract id should be exactly the same. I also feel that the UI entities should have their own model and not be a JSON Field in the Contract model (and yes, this implies to change the react part too). With those changes you can achieve pretty easily a one to one relationship between the UI instances and their respective Kernel instances.

Reply:
This is what I mean

```text

    +------------------+       +------------------+       +------------------+
    | Pipeline         |       | Contract         |       | Entity           |
    +==================+       +==================+       +==================+
    | id               |<--+   | id               |<--+   | id               |
    | name             |   |   | name             |   |   | name             |
    | schema           |   |   | rules            |   |   | definition       |
    | input            |   |   | ...              |   |   | ...              |
    | mappingset_id    |   |   | mapping_id       |   |   | schema_id/ps_id  |
    +------------------+   |   +::::::::::::::::::+   |   +::::::::::::::::::+
                           +--<| pipeline         |   +--<| contract         |
                               +------------------+       +------------------+
```

You don't need any `kernel_refs` anymore because those values are saved in each object.

What do you think?

Reply:
This will help especially for querying, but I am of the option that since we now have a kernel based `project/artefacts` generation function, ui configuration of these artefacts should be reduced to the barest minimum, while the artefacts generator be made more flexible and robust. So some other 3rd part apps and consume it with minimum configurations